tmt/steps/provision/beaker_mrack.py

```diff
@@ -0,0 +1,532 @@
+                f"Failed to create, response: '{response}'.")
+
+        self.guestname = response["id"] if not response["system"] else response["system"]
+        self.info('guestname', self.guestname, 'green')
+
+        with updatable_message(
+                "status", indent_level=self._level()) as progress_message:
+
+            def get_new_state() -> GuestInspectType:
+                response = self.api.inspect()
+
+                if response["status"] == "Aborted":
+                    raise ProvisionError(
+                        f"Failed to create, "
+                        f"unhandled API response '{response['status']}'."
+                        )
+
+                current = cast(GuestInspectType, response)
+                state = current["status"]
+                state_color = GUEST_STATE_COLORS.get(
+                    state, GUEST_STATE_COLOR_DEFAULT
+                    )
+
+                progress_message.update(state, color=state_color)
+
+                if state in {"Error, Aborted", "Cancelled"}:
+                    raise ProvisionError(
+                        'Failed to create, provisioning failed.'
+                        )
+
+                if state == 'Reserved':
+                    return current
+
+                raise tmt.utils.WaitingIncomplete()
+
+            try:
+                guest_info = tmt.utils.wait(
+                    self, get_new_state, datetime.timedelta(
+                        seconds=self.provision_timeout),
+                    tick=self.provision_tick
+                    )
+
+            except tmt.utils.WaitingTimedOutError:
+                response = self.api.delete()
+                raise ProvisionError(
+                    f'Failed to provision in the given amount '
+                    f'of time (--provision-timeout={self.provision_timeout}).'
+                    )
+
+        self.guest = guest_info['system']
+        self.info('address', self.guest, 'green')
+
+    def start(self) -> None:
+        """
+        Start the guest
+
+        Get a new guest instance running. This should include preparing
+        any configuration necessary to get it started. Called after
+        load() is completed so all guest data should be available.
+        """
+
+        if self.guestname is None or self.guest is None:
+            self._create()
+
+    def remove(self) -> None:
+        """ Remove the guest """
+
+        if self.guestname is None:
+            return
+
+        self.api.delete()
+
+
+@tmt.steps.provides_method('beaker')
+class ProvisionBeaker(tmt.steps.provision.ProvisionPlugin):
+    """
+    Provision guest on Beaker system using mrack
+
+    Minimal configuration could look like this:
+
+        provision:
+            how: beaker
+            image: Fedora
+
+    Full configuration example: TBA
+
+    """
+
+    _data_class = ProvisionBeakerData
+    _guest_class = GuestBeaker
+
+    # Guest instance
+    _guest = None
+
+    _keys = [
```

Comment:
`_keys` are not needed anymore, `ProvisionBeakerData` defines them already.

Reply:
removed

Reply:
yet the list still appears in the diff...

Reply:
hehe removed the other part :D let me try to fix